pkg/kube/util/mutate/mutate.go

```diff
@@ -71,7 +71,7 @@ func ESecMutateFn(eSec *esv1.ExtendedSecret) controllerutil.MutateFn {
 		eSec.Labels = updated.Labels
 		eSec.Annotations = updated.Annotations
 		// Update only when spec has been changed
-		if !reflect.DeepEqual(eSec.Spec, updated.Spec) {
+		if !reflect.DeepEqual(eSec.Spec, updated.Spec) || eSec.Status.Generated != updated.Status.Generated {
```

Comment:
this could be two separate `if`s. There is no reason to copy spec, if just status changed and vice versa.

Reply:
Done.